### DISCUSSION TITLE
Why do you have to manually defiine the environment ariables when using as a login shell.

### DISCUSSION THREAD

**Discussion by maltalef101** [nushell/nushell#16664] [Q&A]
### Question

Why can't Nushell do like every other shell does and source them beforehand? Be it sourcing `/etc/profile` or what else. This is no kind of issue for bash or even dash to deal with, if I need some environment variables, then I define them in a profile file.

On NixOS it seems I have two places I should define environment variables in: `configuration.nix`'s `enviroment.variables = {};`, and Nushells `env.nu` file, if I want it to be my login shell.

I find Nushell a great shell, but this is certainly an annoyance.

Regards.

**Comment by ParadoxZero** [comment]
The login esque stuff seems handled on it's own for me when using nushell as login shell. 
Things like SSH X11 forwarding or other system env are populated correctly.

I am using ubuntu server.
